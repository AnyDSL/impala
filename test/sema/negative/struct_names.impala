struct S {
    a : i32
}
struct T {
    a : i32
}

fn getT(a: i32) -> T { getS(a) }
fn getS(a: i32) -> S { S { a : a } }

fn main(i: i32) -> i32 {
    getT(i).a
}
