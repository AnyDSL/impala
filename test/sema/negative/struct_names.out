struct_names.impala:8 col 22 - 32: error: mismatched types: expected 'T' but found 'S' as return type
